src/components/TextInput/BaseTextInput.js

```diff
@@ -274,6 +273,7 @@ class BaseTextInput extends Component {
                                         onPressOut={this.props.onPress}
                                         showSoftInputOnFocus={!this.props.disableKeyboard}
                                         keyboardType={getSecureEntryKeyboardType(this.props.keyboardType, this.props.secureTextEntry, this.state.passwordHidden)}
+                                        value={this.props.value}
```

Comment:
Hmm, @eVoloshchak did you test on the production build for Android?

My cursor is in the middle and holding backspace deletes all the numbers

https://user-images.githubusercontent.com/29673073/182021683-000a213d-8c8e-48a6-a8c7-3a2e20467d7f.mp4

Reply:
> Hmm, @eVoloshchak did you test on the production build for Android?
Weird, I remember this behavior being present only on DEV. And it seemed to be the same for [7433](https://github.com/Expensify/App/pull/7433#issuecomment-1032805299). But now i'm getting this bug with about 60% chance.
This is happening because when user holds `<`, [updateAmountNumberPad](https://github.com/Expensify/App/blob/a06358cef3ab1951fcbb5c256c003acdf238d447/src/pages/iou/steps/IOUAmountPage.js#L146) is called every 100 ms. But due to some kind of event bubbling issue, textinput's [onSelectionChange](https://github.com/Expensify/App/blob/df3b5f5b4ca6a44c6ab21cb4e55d3f0a43e6c992/src/pages/iou/steps/IOUAmountPage.js#L256) method is called with `selection` object being incorrect, probably due to poor performance, since it intencifies in `DEV` mode. Since we're controlling selection manually in `updateAmountNumberPad` method, we don't need the selection data from `onSelectionChange` when user is holding the `<` button
1. We can add a local variable in IOUAmountPage
```javascript
this.shouldUpdateSelection = true;
```
2. Add a new method to `BigNumberPad`, which will be called when user starts and stops long pressing
``` diff
    handleLongPress(key) {
        // Only handles deleting.
        if (key !== '<') {
            return;
        }
+       this.props.longPressHandlerStateChanged(true);
        const timer = setInterval(() => {
            this.props.numberPressed(key);
        }, 100);
        this.setState({timer});
    }
```

``` diff
  onPressOut={() => {
      clearInterval(this.state.timer);
      ControlSelection.unblock();
      this.props.longPressHandlerStateChanged(false);
  }}
```
3. And then on IOUAmountPage we can ignore `onSelectionChange`  if user is long pressing
```diff
  <BigNumberPad
      numberPressed={this.updateAmountNumberPad}
+     longPressHandlerStateChanged={state => this.shouldUpdateSelection = !state}
  />
```
```diff
  onSelectionChange={(e) => {
+    if (!this.shouldUpdateSelection) {
+        return;
+    }
      this.setState({selection: e.nativeEvent.selection});
  }}
```

https://user-images.githubusercontent.com/9059945/182027798-73f76ea1-c73a-463b-a1b4-ff6aa5058c90.mp4